### ISSUE TITLE
Visualising PECUBE Output

### ISSUE THREAD

**Issue by peizenhofer** [JuliaGeodynamics/GeophysicalModelGenerator.jl#12]
Hi Boris & Marcel,

It would be helpful to visualise PECUBE thermal output incl. age prediction on the surface. I could provide our Tübingen specific output files if needed but as you suggested it's probably better to look at the output from J. Braun's original code. 

Alternatively, we formatted our output as Tecplot files. So maybe a general Tecplot file wrapper would be helpful?

Otherwise, really cool work!

Cheers,
Paul

**Comment by boriskaus** [comment]
Seems like a good idea. Could you give us a link to the PECUBE code, and in particular, it's output format? Do you do simulations in cartesian coordinates, or are they in `UTM` or `Lon/Lat`?

**Comment by peizenhofer** [comment]
Link to original PECUBE source code by Jean Braun: 
https://github.com/jeanbraun/Pecube

There you'll find documentation of output formats as well as examples. As mentioned, we've heavily modified J. Braun's version and work with different, Tübingen specific input and output files.

**Comment by mthielma** [comment]
It seems if PECUBE also writes out CSV files. If that is the case, we can handle that relatively easily once we know how it is organized. Handling the VTK files that are apparently also generated should also not be such a big issue. For the binary output files, the PECUBE manual states:
> The binary files exist mostly for legacy reasons. They should not be used by the user. If a user needs access to the complete temperature structure, for example, the author should be contacted for further information on how to access these binary files.

**Comment by boriskaus** [comment]
yes, it would indeed be helpful if you can explain us better how you actually use it, and what would be the ideal requirements. Write topography of a certain location as input for PECUBE?

**Comment by peizenhofer** [comment]
One problem with us (Tübingen in particular, and probably many other thermochronology groups) using PECUBE is that input/output files are not standardised, somewhat based on J. Braun's original code. Perhaps a better solution would be if you provide a general output format that is best suitable for you and that we in the thermochronology community will follow when using the GeophysicalModelGenerator, say a CSV file with [x,y,z,age1,age2,etc]. I then could provide you with a suitably formatted output of one of my recent Alps 2D sections for testing.

Here in Tübingen we primarily apply PECUBE to 2D sections at the moment (can theoretically extend that to 3D). We produce two primary PECUBE output types: temperature [x, y, z, temperature] and surface age prediction [x,y,surface_elevation,age1,age2,etc]. This general format should be common across the community.

Regarding topography: we usually provide a [x,y,z] user topography file as input sometimes tied to an UTM coordinate. This is not necessarily the present-day surface topography because our topography is often based on external topography modelling.

**Comment by boriskaus** [comment]
Could you perhaps write a small Julia scripts that reads in your CSV file? Have a look at the first [example](https://juliageodynamics.github.io/GeophysicalModelGenerator.jl/dev/man/tutorial_load3DSeismicData/). 
We could take it from there.

**Comment by mthielma** [comment]
Great thanks, I will have a look! Do you want to make a tutorial out of it that we can put in the documentation? I can also help you with that.

**Comment by peizenhofer** [comment]
As you will see, it's an extremely simple script (and only works for Tübingen specific PECUBE output). I'm not sure if it warrants a dedicated tutorial page. Perhaps by adding longitude/latitude coordinate conversion and creation of a Paraview file (similar to the example page that Boris linked) could work as an individual tutorial page though. Let me know.
